From cb657871f2dd2d3196fb05fe8fdd4b77cfa9bac6 Mon Sep 17 00:00:00 2001
From: Ben Cressey <bcressey@amazon.com>
Date: Wed, 11 Mar 2026 00:02:05 +0000
Subject: [PATCH] disable RPC helper

We do not intend to run the daemon to synchronize firewall rules or
to collect statistics, so we do not need its protocol helpers.

The RPC helper would require us to package libtirpc, which we want
to avoid for now since we have no other need for it.

Signed-off-by: Ben Cressey <bcressey@amazon.com>
[KCSesh:
  - rebased to 1.4.9]
Signed-off-by: Kyle Sessions <kssessio@amazon.com>

---
 configure.ac            | 2 --
 src/helpers/Makefile.am | 5 -----
 2 files changed, 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index a72963c..eb032a7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -49,8 +49,6 @@ AC_ARG_ENABLE([systemd],
         AS_HELP_STRING([--enable-systemd], [Build systemd support]),
         [enable_systemd="$enableval"], [enable_systemd="no"])

-AC_CHECK_HEADER([rpc/rpc_msg.h], [AC_SUBST([LIBTIRPC_CFLAGS],'')], [PKG_CHECK_MODULES([LIBTIRPC], [libtirpc])])
-
 PKG_CHECK_MODULES([LIBNFNETLINK], [libnfnetlink >= 1.0.1])
 PKG_CHECK_MODULES([LIBMNL], [libmnl >= 1.0.3])
 PKG_CHECK_MODULES([LIBNETFILTER_CONNTRACK], [libnetfilter_conntrack >= 1.1.1])
diff --git a/src/helpers/Makefile.am b/src/helpers/Makefile.am
index e458ab4..47f1d46 100644
--- a/src/helpers/Makefile.am
+++ b/src/helpers/Makefile.am
@@ -4,7 +4,6 @@ pkglib_LTLIBRARIES = ct_helper_amanda.la \
 		     ct_helper_dhcpv6.la \
 		     ct_helper_ftp.la	\
 		     ct_helper_mdns.la	\
-		     ct_helper_rpc.la	\
 		     ct_helper_tftp.la	\
 		     ct_helper_tns.la	\
 		     ct_helper_sane.la	\
@@ -30,10 +29,6 @@ ct_helper_mdns_la_SOURCES = mdns.c
 ct_helper_mdns_la_LDFLAGS = $(HELPER_LDFLAGS)
 ct_helper_mdns_la_CFLAGS = $(HELPER_CFLAGS)

-ct_helper_rpc_la_SOURCES = rpc.c
-ct_helper_rpc_la_LDFLAGS = $(HELPER_LDFLAGS)
-ct_helper_rpc_la_CFLAGS = $(HELPER_CFLAGS) $(LIBTIRPC_CFLAGS)
-
 ct_helper_tftp_la_SOURCES = tftp.c
 ct_helper_tftp_la_LDFLAGS = $(HELPER_LDFLAGS)
 ct_helper_tftp_la_CFLAGS = $(HELPER_CFLAGS)

